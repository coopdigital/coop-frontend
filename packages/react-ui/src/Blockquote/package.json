{
  "name": "@coopdigital/blockquote",
  "version": "1.0.2",
  "description": "Co-op Components: Blockquote",
  "main": "dist/index.js",
  "style": "dist/blockquote.css",
  "author": "@coopdigital",
  "contributors": [
    "Michael Chadwick <michael.chadwick@coop.co.uk> (https://github.com/mchadwickweb)",
    "Dale Hudson <dale.hudson@coop.co.uk> (https://github.com/dalecreativ)"
  ],
  "private": false,
  "files": [
    "dist",
    "src"
  ],
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/coopdigital/coop-frontend.git",
    "directory": "packages/component-blockquote"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^5.11.9",
    "@testing-library/react": "^11.2.5",
    "prop-types": "^15.5.4",
    "react": "^16.14.0",
    "react-dom": "^16.14.0"
  },
  "peerDependencies": {
    "prop-types": "^15.5.4",
    "react": "^16.14.0",
    "react-dom": "^16.14.0"
  },
  "bugs": {
    "url": "https://github.com/coopdigital/coop-frontend/issues"
  },
  "homepage": "https://github.com/coopdigital/coop-frontend/tree/master/packages/component-blockquote#readme",
  "keywords": [
    "blockquote"
  ],
  "publishConfig": {
    "access": "public"
  },
  "gitHead": "f58fae660bb8240497376af98b0ed2bc9e0fd47d"
}
