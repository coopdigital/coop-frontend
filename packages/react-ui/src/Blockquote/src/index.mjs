import Blockquote from './Blockquote.jsx';

export default Blockquote;
