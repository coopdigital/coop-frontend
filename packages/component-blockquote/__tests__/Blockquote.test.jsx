/**
 * @jest-environment jsdom
 */

import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Blockquote from '../src/index.mjs';

describe('Blockquote', () => {
  it('should render default Blockquote', () => {
    const wrapper = mount(<Blockquote citation="citation one" quote="this is a quote and is required"/>);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<Blockquote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should work with all variants', () => {
    const Variants = () => (
      <div>
        <Blockquote quoteLarge citation="cite" quote="this is a quote and is required"/>
      </div>
    );

    const wrapper = mount(<Variants />);
    expect(wrapper.find('.coop-t-blockquote p').hasClass('coop-t-blockquote__quote--large')).toEqual(true);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<Variants />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
