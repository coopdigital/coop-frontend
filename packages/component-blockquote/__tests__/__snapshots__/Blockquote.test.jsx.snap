// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote should render default Blockquote 1`] = `
<blockquote
  className="coop-t-blockquote"
>
  <p
    className="coop-t-blockquote__quote"
  >
    this is a quote and is required
  </p>
  <cite
    className="coop-t-blockquote__cite"
  >
    citation one
  </cite>
</blockquote>
`;

exports[`Blockquote should work with all variants 1`] = `
<div>
  <blockquote
    className="coop-t-blockquote"
  >
    <p
      className="coop-t-blockquote__quote coop-t-blockquote__quote--large"
    >
      this is a quote and is required
    </p>
    <cite
      className="coop-t-blockquote__cite"
    >
      cite
    </cite>
  </blockquote>
</div>
`;
